mesonbuild/backend/backends.py

```diff
@@ -298,13 +298,30 @@ def determine_linker(self, target):
             raise MesonException(m.format(target.name))
         return l
 
+    def rpaths_for_bundled_shared_libraries(self, target):
+        paths = []
+        for dep in target.external_deps:
+            if isinstance(dep, dependencies.ExternalLibrary):
+                la = dep.link_args
+                if len(la) == 1 and la[0].startswith(self.environment.get_source_dir()):
```

Comment:
Can we just do this for all external libraries with a `dirs:` kwarg? It would solve #314 too.

Reply:
I don't have a good grasp on the potential breakage of that change. Rpath has a maximum length IIRC and just shoving everything there could break things. If we ever want to undo that change it will break a lot of things. Maybe start with this, see how it is being used and expand later?